pkg/kubernetes/container/v1alpha1/container.go

```diff
@@ -205,3 +274,103 @@ func WithVolumeMounts(volumeMounts []corev1.VolumeMount) OptionFunc {
 		c.VolumeMounts = volumeMounts
 	}
 }
+
+// WithImagePullPolicy sets the image pull policy of the container
+func (b *Builder) WithImagePullPolicy(policy corev1.PullPolicy) *Builder {
+	if len(policy) == 0 {
+		b.errors = append(
+			b.errors,
+			errors.New(
+				"failed to build container object: missing imagepullpolicy",
+			),
+		)
+		return b
+	}
+
+	b.con.ImagePullPolicy = policy
+	return b
+}
+
+// WithPrivilegedSecurityContext sets securitycontext of the container
+func (b *Builder) WithPrivilegedSecurityContext(privilaged *bool) *Builder {
+	if privilaged == nil {
+		b.errors = append(
+			b.errors,
+			errors.New(
+				"failed to build container object: missing securitycontext",
+			),
+		)
+		return b
+	}
+
+	newprivilaged := *privilaged
+	b.con.SecurityContext.Privileged = &newprivilaged
+	return b
+}
+
+// WithResources sets resources of the container
+func (b *Builder) WithResources(
+	resources *corev1.ResourceRequirements,
+) *Builder {
+	if resources == nil {
+		b.errors = append(
+			b.errors,
+			errors.New("failed to build container object: missing resources"),
+		)
+		return b
+	}
+
+	newresources := *resources
+	b.con.Resources = newresources
+	return b
+}
+
+// WithPorts sets ports of the container
+func (b *Builder) WithPorts(ports []corev1.ContainerPort) *Builder {
```

Comment:
Does this merge or result into duplicates. UT is needed.

Reply:
Renamed it to WithPortsNew as it resets the field. Added UT for all new methods